pastas/stats/signatures.py

```diff
@@ -1008,170 +1331,206 @@ def _baseflow(series: Series, normalize: bool = True) -> Tuple[Series, Series]:
+    c = cos(thetas).sum()
+    s = sin(thetas).sum()
+
+    if (s > 0) & (c > 0):
+        mean_theta = arctan(s / c)
+    elif c < 0:
+        mean_theta = arctan(s / c) + pi
+    elif (s < 0) & (c > 0):
+        mean_theta = arctan(s / c) + two_pi
+    else:
+        # This should never happen
+        raise ValueError("Something went wrong in the circular statistics.")
 
-    Notes
-    -----
-    Test statistic of the dip test; maximum distance between the empirical
-    distribution and the best fitting unimodal distribution. By default, the best
-    fitting distribution is the uniform.
+    return mean_theta * 365.25 / two_pi
 
-    """
-    return NotImplementedError
 
+def date_min(series: Series) -> float:
+    """Compute the average date of the minimum head value with circular statistics.
 
-def critical_bandwidth(series: Series):
-    """Test statistic of the Silverman test, after :cite:t:`silverman_using_1981`.
+    Parameters
+    ----------
+    series: pandas.Series
+        Pandas Series with DatetimeIndex and head values.
 
     Returns
     -------
+    float:
+        Average date of the minimum head value.
 
     Notes
     -----
-    Test statistic of the Silverman test; minimum kernel bandwidth required to create
-    an unimodal distribution estimated by fitting a Kernel Density Estimation.
+    Average date of the minimum head value. The higher the date, the later the minimum
+    head value occurs in the year, and vice versa.
+
+    The average date is computed by taking the average of the day of the year of the
+    minimum head value for each year, using circular statistics. We refer to
+    :cite:t:`fisher_statistical_1995` (page 31) for more information on circular
+    statistics.
 
     """
-    return NotImplementedError
+    return _date_min_max(series, "min")
 
 
-def peak_base_time(series: Series):
-    """Difference between peak and base head, standardized by duration of peak after
-    :cite:t:`heudorfer_index-based_2019`.
+def date_max(series: Series) -> float:
+    """Compute the average date of the maximum head value with circular statistics.
+
+    Parameters
+    ----------
+    series: pandas.Series
+        Pandas Series with DatetimeIndex and head values.
 
     Returns
     -------
+    float:
+        Average date of the maximum head value.
 
     Notes
     -----
-    Difference between peak and base head, standardized by duration of peak.
+    Average date of the maximum head value. The higher the date, the later the maximum
+    head value occurs in the year, and vice versa.
+
+    The average date is computed by taking the average of the day of the year of the
+    maximum head value for each year, using circular statistics. We refer to
+    :cite:t:`fisher_statistical_1995` (page 31) for more information on circular
+    statistics.
 
     """
-    return NotImplementedError
+    return _date_min_max(series, "max")
 
 
 def summary(series: Series, signatures: Optional[list] = None) -> Series:
```

Comment:
It is very likely that one of the checks will fail when all checks are ran. It would be nice to have an option to return `NaN` or `None` for that signature instead of raising an error. This could be done by adding an argument to the `summary` function similar to the `errors` argument in pandas `to_numeric` function.